### PR TITLE
VMO-2926 add duplicate block action

### PR DESCRIPTION
**Pull request by safydy-r**

### CODE REVIEW

**Review by jcbashdown — DISMISSED** [review]
Looks good to me!

**Comment by jcbashdown** [comment]
Sorry for approving and unapproving. Noticed a potential bug here - I'm unable to connect all of the exits on duplicate blocks. I know that this is buggy in general but something I've noticed here specifically - duplicating the select one response block at http://localhost:8080/trees/1/interaction-designer/edit I find that I can't link the existing two exits (exits already in the original block) in the duplicate BUT, If I delete the exits before duplicating I can connect them in the duplicate.

**Comment by safydy-r** [comment]
> Sorry for approving and unapproving. Noticed a potential bug here - I'm unable to connect all of the exits on duplicate blocks. I know that this is buggy in general but something I've noticed here specifically - duplicating the select one response block at http://localhost:8080/trees/1/interaction-designer/edit I find that I can't link the existing two exits (exits already in the original block) in the duplicate BUT, If I delete the exits before duplicating I can connect them in the duplicate.

**Comment by safydy-r** [comment]
And weird, I wanted to comment and made a mistake. I closed the PR instead :|

File: src/store/flow/flow.ts [hunk 1]
```diff
@@ -215,7 +215,43 @@ export const actions: ActionTree<IFlowsState, IRootState> = {
...
       }),
     }
-  }
+  },
+
+  async flow_duplicateBlock({dispatch, commit, state}, {flowId, blockId}: {flowId: string, blockId: IBlock['uuid']}): Promise<IBlock> {
+    const flow = findFlowWith(flowId || state.firstFlowId || '', state as unknown as IContext)
+    const block: IBlock = findBlockWith(blockId, flow)  // @throws ValidationException when block absent
+
+    console.log(JSON.stringify(block))
```
Comment (jcbashdown) [inline src/store/flow/flow.ts:224]: I think we should delete this?

**Review by jcbashdown — APPROVED** [review]
Looks good I think! Just one small thing to fix

File: src/store/flow/flow.ts (hunk 1 above)
Comment (safydy-r) [reply src/store/flow/flow.ts:224]: thnx, missed this one

**Pull request merged**